lib/core/src/server/preview/iframe-webpack.config.js

```diff
@@ -52,18 +54,25 @@ export default ({
     const match = entryFilename.match(/(.*)-generated-(config|other)-entry.js$/);
     if (match) {
       const configFilename = match[1];
+      // This code is NOT transpiled, must be ES5!
       virtualModuleMapping[entryFilename] = `
         import { addDecorator, addParameters, addParameterEnhancer } from '@storybook/client-api';
         import { logger } from '@storybook/client-logger';
 
-        const { decorators, parameters, parameterEnhancers, globalArgs, globalArgTypes, args, argTypes } = require(${JSON.stringify(
-          configFilename
-        )});
+        var m = require(${JSON.stringify(configFilename)});
 
-        if(args || argTypes) logger.warn('Invalid args/argTypes in config, ignoring.', JSON.stringify({ args, argTypes }));
-        if (decorators) decorators.forEach(decorator => addDecorator(decorator));
-        if (parameters || globalArgs || globalArgTypes) addParameters(Object.assign({}, parameters, { globalArgs, globalArgTypes }));
-        if (parameterEnhancers) parameterEnhancers.forEach(enhancer => addParameterEnhancer(enhancer));
+        if (m.args || m.argTypes) {
+          logger.warn('Invalid args/argTypes in config, ignoring.', JSON.stringify(m));
+        }
+        if (m.decorators) {
+          m.decorators.forEach(function(decorator){ return addDecorator(decorator); });
+        }
+        if (m.parameters || m.globalArgs || m.globalArgTypes) {
+          addParameters(Object.assign({}, m.parameters, { globalArgs: m.globalArgs, globalArgTypes: m.globalArgTypes }));
+        }
+        if (m.parameterEnhancers) {
+          m.parameterEnhancers.forEach(function(enhancer){ return addParameterEnhancer(enhancer)});
+        }
       `;
```

Comment:
This code should be in a separate file.

If we do something like this:
```
      virtualModuleMapping[entryFilename] = require('fs-extra')
        .readFileSync('./virtualModule.js')
        .replace('__FOO__BAR__', JSON.stringify(configFilename));
```

We can not only get syntax highlighting, eslint, autocomplete etc on it.. but we can also get to use ES6 in this file.

Reply:
Great idea ❤️

Reply:
Done, I let you check !